### PR TITLE
Add network recovery interval amqp config

### PR DESCRIPTION
**Pull request by nsatvik**

### CODE REVIEW

**Comment by ganeshs** [comment]
@nsatvik travis pipeline broke. Can you check please

File: gojek-commons-amqp/src/main/java/com/gojek/amqp/AmqpConfiguration.java [hunk 1]
```diff
@@ -21,29 +25,40 @@
 	private Integer minChannels = 5;
 	
 	private Integer maxIdleChannels = 5;
-	
+
+	private Integer networkRecoveryInterval = 1000;
+
+	private List<String> hosts = Lists.newArrayList();
+
+	private Address[] addresses;
```
Comment (ganeshs) [inline gojek-commons-amqp/src/main/java/com/gojek/amqp/AmqpConfiguration.java]: Can you mark this as JsonIgnore

File: gojek-commons-amqp/src/test/java/com/gojek/amqp/AmqpConnectionManagerTest.java [hunk 2]
```diff
@@ -38,7 +40,7 @@ public void setup() throws Exception {
 		ConnectionFactory factory = mock(ConnectionFactory.class);
 		doReturn(factory).when(manager).createConnectionFactory();
 		connection = mock(Connection.class);
-		when(factory.newConnection()).thenReturn(connection);
+		when(factory.newConnection(new Address[]{})).thenReturn(connection);
```
Comment (ganeshs) [inline gojek-commons-amqp/src/test/java/com/gojek/amqp/AmqpConnectionManagerTest.java:44]: Also add a test for ensuring setNetworkRecoveryInterval works in the factory

File: gojek-commons-amqp/src/main/java/com/gojek/amqp/AmqpConfiguration.java [hunk 3]
```diff
@@ -21,29 +26,41 @@
 	private Integer minChannels = 5;
 	
 	private Integer maxIdleChannels = 5;
-	
+
+	private Integer networkRecoveryInterval = 1000;
+
+	private List<String> hosts = Lists.newArrayList();
```
Comment (khajurkar) [inline gojek-commons-amqp/src/main/java/com/gojek/amqp/AmqpConfiguration.java:32]: Initialisation is superfluous. No need of pulling google dependencies just for this.

**Pull request merged**